katdal/chunkstore_s3.py

```diff
@@ -63,6 +63,11 @@ class S3ChunkStore(ChunkStore):
     def __init__(self, client):
         if not botocore:
             raise _botocore_import_error
+        # XXX Hack to route head_object 404 error to ChunkNotFound via
+        # NoSuchKey. This is better than routing all ClientErrors to
+        # ChunkNotFound, but the best solution would be a fix to botocore.
+        boto_error_map = client.exceptions._code_to_exception
+        boto_error_map['404'] = boto_error_map['NoSuchKey']
```

Comment:
In case you wanted something that doesn't reach into the class internals, here's an alternative solution: catch the ClientError and check the error code: https://stackoverflow.com/questions/33842944/check-if-a-key-exists-in-a-bucket-in-s3-using-boto3

Reply:
Done.